composer.json

```diff
@@ -22,12 +22,16 @@
 		"phpcompatibility/phpcompatibility-wp": "^2.1",
 		"squizlabs/php_codesniffer": "^3.5",
 		"wp-coding-standards/wpcs": "^2.3.0",
-		"yoast/wp-test-utils": "^1"
+		"yoast/wp-test-utils": "^1",
+		"phpstan/phpstan": "^1.3",
+		"phpstan/extension-installer": "^1.1",
```

Comment:
https://github.com/ergebnis/phpstan-rules might be another good extension to look into - a good selection of extra rules.

Reply:
I'm not sure on this one. Rules are way more strict, and when running it we get plenty of errors for using things like `isset` or `empty`. I can see the point on not using them, but I'm not sure up until which point those are real issues or noise.

Reply:
We don't have to use them straight away - but we can work towards some of them over time.

https://github.com/phpstan/phpstan-strict-rules would be another set to consider to improve the overall codebase.

Reply:
I've actually tried this last proposal and I like it. It will give us more work to clean all the errors up, but we can do that over time.

Reply:
https://phpstan.org/user-guide/discovering-symbols